sysl2/sysl/sysl_test.go

```diff
@@ -211,32 +203,47 @@ func TestMain2JSONStdout(t *testing.T) {
 }
 
 func TestMain2BadMode(t *testing.T) {
-	var stdout, stderr bytes.Buffer
-	rc := main2(&stdout, &stderr, []string{"sysl", "-o", "-", "-mode", "BAD", "tests/args.sysl"}, main3)
+	rc := main2([]string{"sysl", "-o", "-", "-mode", "BAD", "tests/args.sysl"}, main3)
 	assert.NotZero(t, rc)
 
 	_, err := os.Stat("-")
 	assert.True(t, os.IsNotExist(err), "Should not have created file '-'")
 }
 
 func TestMain2BadLog(t *testing.T) {
-	var stdout, stderr bytes.Buffer
-	rc := main2(&stdout, &stderr, []string{"sysl", "-o", "-", "-log", "BAD", "tests/args.sysl"}, main3)
+	rc := main2([]string{"sysl", "-o", "-", "-log", "BAD", "tests/args.sysl"}, main3)
 	assert.NotZero(t, rc)
 
 	_, err := os.Stat("-")
 	assert.True(t, os.IsNotExist(err), "Should not have created file '-'")
 }
 
+func TestMain2WithBlackboxParams(t *testing.T) {
+	testHook := test.NewGlobal()
+	main2([]string{"sd", "-s", "MobileApp <- Login", "-o", "tests/call.png", "-b", "Server <- DB=call to database",
+		"-b", "Server <- Login=call to database", "tests/call.sysl"}, main3)
+	assert.Equal(t, logrus.WarnLevel, testHook.LastEntry().Level)
+	assert.Equal(t, "blackbox 'Server <- DB' not hit in app ' Global :: '\n", testHook.LastEntry().Message)
```

Comment:
blackbox `Server <- DB` is at endpoint collection and not at app level.
if -o has `%(epname)` then all blackboxes are at app level.

Reply:
remove `' Global :: '` from the output,